app/Http/Controllers/StudentEnrollmentController.php

```diff
@@ -0,0 +1,356 @@
+            ->first();
+
+        if ($existingEnrollment) {
+            return response()->json([
+                'message' => 'Student already enrolled in this course'
+            ], 409);
+        }
+
+        // LOGIKA BARU - Assign dosen secara acak dan merata
+        $assignedLecturerId = $this->assignLecturerToStudent($request->course_id);
+
+        if (!$assignedLecturerId) {
+            return response()->json([
+                'message' => 'No lecturers available for this course'
+            ], 400);
+        }
+
+        $enrollment = StudentEnrollment::create([
+            'user_id' => $request->user_id,
+            'course_id' => $request->course_id,
+            'semester' => $request->semester,
+            'tahun_akademik' => '2024/2025',
+            'status' => 'active',
+            'lecturer_id' => $assignedLecturerId  // TAMBAHAN BARU
+        ]);
+
+        $enrollment->load(['course', 'course.lecturers', 'lecturer']);
+
+        // Transform response data
+        $transformedEnrollment = [
+            'id' => (string) $enrollment->id,
+            'user_id' => (string) $enrollment->user_id,
+            'course_id' => (string) $enrollment->course_id,
+            'semester' => $enrollment->semester,
+            'tahun_akademik' => $enrollment->tahun_akademik,
+            'status' => $enrollment->status,
+            'lecturer_id' => (string) $enrollment->lecturer_id,  // TAMBAHAN BARU
+            'assigned_lecturer' => $enrollment->lecturer->nama,   // TAMBAHAN BARU
+            'created_at' => $enrollment->created_at->toISOString(),
+            'updated_at' => $enrollment->updated_at->toISOString(),
+            'course' => [
+                'id' => (string) $enrollment->course->id,
+                'kode' => $enrollment->course->kode,
+                'nama' => $enrollment->course->nama,
+                'semester' => $enrollment->course->semester,
+                'sks' => $enrollment->course->sks,
+                'lecturers' => $enrollment->course->lecturers->map(function ($lecturer) {
+                    return [
+                        'id' => (string) $lecturer->id,
+                        'nama' => $lecturer->nama,
+                        'nip' => $lecturer->nip,
+                    ];
+                })
+            ]
+        ];
+
+        return response()->json([
+            'message' => 'Successfully enrolled in course with assigned lecturer: ' . $enrollment->lecturer->nama,
+            'data' => $transformedEnrollment
+        ], 201);
+
+    } catch (\Illuminate\Validation\ValidationException $e) {
+        return response()->json([
+            'message' => 'Validation failed',
+            'errors' => $e->errors()
+        ], 422);
+    } catch (\Exception $e) {
+        Log::error('Error enrolling in course: ' . $e->getMessage());
+        return response()->json([
+            'message' => 'Error enrolling in course',
+            'error' => $e->getMessage()
+        ], 500);
```

Comment:
Error 5xx sebaiknya tidak dibuat hasil errornya in response. 
```suggestion
    } catch (\Exception $e) {
        Log::error('Error enrolling in course: ' . $e->getMessage());
        return response()->json([
            'message' => 'Error enrolling in course',
            'error' => "INTERNAL_SERVER_ERROR"
        ], 500);
```